docs/content/concepts/partitions-schedules-sensors/sensors.mdx

```diff
@@ -721,6 +761,89 @@ def uses_db_connection():
 
+### Monitoring asset partitions with sensors
+
+#### Updating a weekly asset partition when upstream daily partitions are materialized
+
+The following code example monitors an upstream daily-partitioned asset, kicking off materializations of a downstream weekly-partitioned asset whenever a daily partition is materialized and all daily partitions in the week have existing materializations. Every time a daily partition is replaced, the weekly partition will materialize.
+
+<PyObject
+object="MultiAssetSensorEvaluationContext"
+method="all_partitions_materialized"
+/> is a utility method accepts a list of partitions and checks if every provided
+partition has been materialized. This method ignores the cursor, so it searches through
+all existing materializations.
+
+```python file=/concepts/partitions_schedules_sensors/sensors/sensors.py startafter=start_daily_asset_to_weekly_asset endbefore=end_daily_asset_to_weekly_asset
+@multi_asset_sensor(asset_keys=[AssetKey("upstream_daily_asset")], job=weekly_asset_job)
+def trigger_weekly_asset_from_daily_asset(context):
+    materializations_by_partition = context.latest_materialization_records_by_partition(
+        AssetKey("upstream_daily_asset")
+    )
+
+    # Get all corresponding weekly partitions for any materialized daily partitions
+    for partition, materialization in materializations_by_partition.items():
+        weekly_partitions = context.get_downstream_partition_keys(
+            partition,
+            from_asset_key=AssetKey("upstream_daily_asset"),
+            to_asset_key=AssetKey("downstream_weekly_asset"),
+        )
+
+        if weekly_partitions:  # Check that a downstream weekly partition exists
+            # Upstream daily partition can only map to at most one downstream weekly partition
+            daily_partitions_in_week = context.get_downstream_partition_keys(
+                weekly_partitions[0],
+                from_asset_key=AssetKey("downstream_weekly_asset"),
+                to_asset_key=AssetKey("upstream_daily_asset"),
+            )
+            if context.all_partitions_materialized(
+                AssetKey("upstream_daily_asset"), daily_partitions_in_week
+            ):
+                yield weekly_asset_job.run_request_for_partition(
+                    weekly_partitions[0], run_key=None
+                )
+                # Advance the cursor so we only check event log records past the cursor
+                context.advance_cursor(
+                    {AssetKey("upstream_daily_asset"): materialization}
+                )
+```
+
+#### Materializing a daily asset partition if both upstream daily partitions are materialized
+
+The following example monitors two upstream daily-partitioned assets, kicking off a run in the downstream daily-partitioned asset if any upstream daily partition is replaced and the other upstream daily partition has an existing materialization.
```

Comment:
I recommend breaking this sentence up into two, at a minimum. It's currently a bit hard to read + absorb due to its length.